app/controllers/middleware_server_controller.rb

```diff
@@ -28,9 +28,69 @@ def listicon_image(item, _view)
     "svg/#{icon}.svg"
   end
 
+  def button
+    super unless %w(middleware_server_reload middleware_server_stop).include?(params[:pressed])
+
+    operation = params[:pressed]
+    items = identify_selected_servers
+
+    return if items.nil? # TODO: re-route somewhere?
+
+    run_server_operation(operation, items)
+
+    render :update do |page|
+      page.replace("flash_msg_div", :partial => "layouts/flash_msg")
+    end
+  end
+
   private ############################
 
   def display_name
     _('Middleware Servers')
   end
+
+  def identify_selected_servers
```

Comment:
We have 2 entry points here, which can be the list view or an individual server. In the former case `:id` holds the id, while in the other case `:miq_grid_checks` holds a list of ids.
I'll add a comment.